### PR TITLE
add logsumexp for pytorch frontend

### LINKED ISSUE #12475: logsumexp

### PR DESCRIPTION
**Pull request by sm1lla**
close #12475

### CODE REVIEW

**Review by YushaArif99 — APPROVED** [review]
Hey @sm1lla 

The PR lgtm! I will now be merging the PR.

Thanks for the contribution! 🙂

File: ivy_tests/test_ivy/test_frontends/test_torch/test_reduction_ops.py [hunk 1]
```diff
@@ -8,8 +8,9 @@
 from ivy_tests.test_ivy.test_functional.test_core.test_statistical import (
     statistical_dtype_values,
 )
-from ivy_tests.test_ivy.test_functional.test_experimental.test_core.test_statistical\
-    import statistical_dtype_values as statistical_dtype_values_experimental
+from ivy_tests.test_ivy.test_functional.test_experimental.test_core.test_statistical import (
```
Comment (YushaArif99) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_reduction_ops.py:11]: Actually, just one small thing before I merge the PR, can you kindly fix the lint error that's being caused by this line. I think the original version was correct so you didnt have to change it

**Review by YushaArif99 — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_torch/test_reduction_ops.py (hunk 1 above)
Comment (sm1lla) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_reduction_ops.py:11]: Hi @YushaArif99, 
sure I'll change it to the old version.👍

Comment (sm1lla) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_reduction_ops.py]: The auto-formatter changes the line, so I just committed it without running the formatter. I hope that's ok.

**Review by YushaArif99 — COMMENTED** [review]
Hey @sm1lla 

I think this PR has become adulterated due to the recent `git push --force` on the master branch. You'll now have to close this PR and open a new one. Sorry for the inconvenience 😅

**Pull request closed without merging**